apps/emqx_prometheus/src/emqx_prometheus_ds_builtin_raft.erl

```diff
@@ -0,0 +1,116 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_prometheus_ds_builtin_raft).
+
+%% for bpapi
+-behaviour(emqx_prometheus_cluster).
+
+-include("emqx_prometheus.hrl").
+
+-define(METRIC_PREFIX, <<"emqx_ds_raft_">>).
+
+%% Please don't remove this attribute, prometheus uses it to
+%% automatically register collectors.
+-behaviour(prometheus_collector).
+-export([
+    deregister_cleanup/1,
+    collect_mf/2
+]).
+
+%% `emqx_prometheus' API
+-export([collect/1]).
+
+%% `emqx_prometheus_cluster' API
+-export([
+    fetch_from_local_node/1,
+    fetch_cluster_consistented_data/0,
+    aggre_or_zip_init_acc/0,
+    logic_sum_metrics/0
+]).
+
+%% `emqx_prometheus_cluster' API
+
+fetch_from_local_node(Mode) ->
+    Labels0 = with_node_label(Mode, []),
+    Acc1 = emqx_ds_builtin_raft_metrics:local_dbs(Labels0),
+    Acc = maps:merge(emqx_ds_builtin_raft_metrics:local_shards(Labels0), Acc1),
+    {node(), Acc}.
+
+fetch_cluster_consistented_data() ->
+    Acc1 = emqx_ds_builtin_raft_metrics:cluster(),
+    Acc2 = maps:merge(emqx_ds_builtin_raft_metrics:dbs(), Acc1),
+    Acc = maps:merge(emqx_ds_builtin_raft_metrics:shards(), Acc2),
+    Acc.
+
+aggre_or_zip_init_acc() ->
+    Meta = local_metrics_meta(),
+    maps:from_keys([element(1, M) || M <- Meta], []).
+
+logic_sum_metrics() ->
+    [].
+
+%%--------------------------------------------------------------------
+%% Collector API
+%%--------------------------------------------------------------------
+
+%% @private
+deregister_cleanup(_) -> ok.
+
+%% @private
+-spec collect_mf(_Registry, Callback) -> ok when
+    _Registry :: prometheus_registry:registry(),
+    Callback :: prometheus_collector:collect_mf_callback().
+collect_mf(?PROMETHEUS_DSRAFT_REGISTRY, Callback) ->
+    Metrics = emqx_prometheus_cluster:raw_data(?MODULE, ?GET_PROM_DATA_MODE()),
+    collect_metric_families(Callback, metrics_meta(), Metrics);
+collect_mf(_, _) ->
+    ok.
+
+%% @private
+collect(<<"json">>) ->
+    #{
+        %% TODO
```

Comment:
Nit: Maybe explicitly state that we are not going to implement this and return empty?